web/docs/project/testing.md

```diff
@@ -1,3 +1,171 @@
+:::
+
+## Testing Your React App
+
+Wasp enables you to quickly and easily write both unit tests and React component tests for your frontend code. Because Wasp uses [Vite](https://vitejs.dev/), we support testing web apps through [Vitest](https://vitest.dev/).
+
+<details>
+  <summary>Included Libraries</summary>
+  <div>
+
+  [`vitest`](https://www.npmjs.com/package/vitest): Unit test framework with native Vite support.
+
+  [`@vitest/ui`](https://www.npmjs.com/package/@vitest/ui): A nice UI for seeing your test results.
+
+  [`jsdom`](https://www.npmjs.com/package/jsdom): A web browser test environment for Node.js.
+
+  [`@testing-library/react`](https://www.npmjs.com/package/@testing-library/react) / [`@testing-library/jest-dom`](https://www.npmjs.com/package/@testing-library/jest-dom): Testing helpers.
+
+  [`msw`](https://www.npmjs.com/package/msw): A server mocking library.
+
+  </div>
+</details>
+
+### Setting Up Tests
+
+In order for Wasp to pick up your tests, they should be placed within the `src/client` directory and use an extension that matches [these glob patterns](https://vitest.dev/config#include). Some of the file names that Wasp will pick up as tests:
+
+- `yourFile.test.ts`
+- `YourComponent.spec.jsx`
+
+Within test files, you can import your other source files as usual. For example, if you have a component `Counter.jsx`, you test it by creating a file in the same directory called `Counter.test.jsx` and import the component with `'./Counter'`.
+
+### Running Tests
+
+Running `wasp test client` will start Vitest in watch mode and recompile your Wasp project when changes are made.
+
+- If you want to see a realtime UI, pass `--ui` as an option.
+- To run the tests just once, use `wasp test client run`.
+
+All arguments after `wasp test client` are passed directly to the Vitest CLI, so check out [their documentation](https://vitest.dev/guide/cli.html) for all of the options.
+
+:::warning
+You should not run `wasp test` while `wasp start` is running. Both will try to compile your project to `.wasp/out`.
+:::
+
+### React Testing Helpers
+
+Wasp provides several functions to help you write React tests:
+
+- `mockQuery`: Takes a Wasp Query to mock and the JSON data to return.
+  - This is helpful if your Query uses `useQuery`.
+  - Behind the scenes, Wasp uses [`msw`](https://npmjs.com/package/msw) to create a server request handler that responds with the given JSON.
+  - Mocks are cleared after each test.
+- `mockApi`: Similar to `mockQuery`, but for [APIs](/docs/advanced/apis) instead.
+  - Instead of a Wasp Query, it takes a route of the shape: `mockApi({ method: HttpMethod.Get, path: '/foo/bar' }, { res: 'hello' })`.
+  - You can import `HttpMethod` like so: `import { HttpMethod } from '@wasp/types'`.
+- `renderInContext`: Takes a React component, wraps it inside a `QueryClientProvider` and `Router`, and renders it.
+
+## Testing Your Server-Side Code
+
+Coming soon!
```

Comment:
I would maybe expand this to say something "We currently don't support testing your server-side code, but we will be adding support soon. You can track the progress [here](https://github.com/wasp-lang/wasp/issues/110) and express your interest by commenting on the Github issue.`